core/src/main/java/cucumber/runtime/formatter/TimelineFormatter.java

```diff
@@ -0,0 +1,265 @@
+package cucumber.runtime.formatter;
+
+import cucumber.api.TestCase;
+import cucumber.api.event.EventHandler;
+import cucumber.api.event.EventPublisher;
+import cucumber.api.event.TestCaseFinished;
+import cucumber.api.event.TestCaseStarted;
+import cucumber.api.event.TestRunFinished;
+import cucumber.api.event.TestSourceRead;
+import cucumber.api.formatter.Formatter;
+import cucumber.api.formatter.NiceAppendable;
+import cucumber.runtime.CucumberException;
+import cucumber.runtime.io.URLOutputStream;
+import gherkin.deps.com.google.gson.Gson;
+import gherkin.deps.com.google.gson.GsonBuilder;
+import gherkin.deps.com.google.gson.annotations.SerializedName;
+import gherkin.pickles.PickleTag;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.net.URL;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class TimelineFormatter implements Formatter {
```

Comment:
this will probably need to implement `ConcurrentEventListener` otherwise all the Thread Ids will be the same (main thread) and the start/end times will be incorrect as they will be when the events are flushed rather than when initially published

note: cannot use the time of the event is nano time but the render of the report requires millis

Reply:
Thx. I'll Look at `TimeUnit`.

Reply:
Oh, I left it using System.currentTimeMillis, meant to change that after my own testing